File: javascript/MDC.js
var input = require('fs').readFileSync('javascript/stdin', 'utf8');
//ESTUDAR
var sep = [' ', '\n'];
var line = input.split(new RegExp(sep.join('|'), 'g'));

let a = 0, b = 0, resto = 0;

console.log("Primeiro numero");
a = parseInt(line.shift());
console.log(a)
console.log("Segundo numero");
b = parseInt(line.shift());
console.log(b)

while (b != 0) {
    resto = a % b;
    a = b;
    b = resto;
}
console.log("O MDC eh: " + a);

File: java/src/MDC.java
import java.util.Scanner;

public class MDC {
    public void calculaMDC() {
        int a = 0, b = 0, resto = 0;
        Scanner input = new Scanner(System.in);

        System.out.println("Primeiro numero");
        a = input.nextInt();
        System.out.println("Segundo numero");
        b = input.nextInt();

        while (b != 0) {
            resto = a % b;
            a = b;
            b = resto;
        }
        System.out.println("O MDC eh: " + a);
    }
}